templates/repo/issue/view_content/show_role.tmpl

```diff
@@ -1,15 +1,27 @@
 {{if and (.ShowRole.HasRole "Poster") (not .IgnorePoster)}}
-	<div class="ui basic label role-label">
-		{{ctx.Locale.Tr "repo.issues.poster"}}
-	</div>
-{{end}}
-{{if (.ShowRole.HasRole "Writer")}}
-	<div class="ui basic label role-label">
-		{{ctx.Locale.Tr "repo.issues.collaborator"}}
+	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.author_helper"}}">
+		{{ctx.Locale.Tr "repo.issues.author"}}
 	</div>
 {{end}}
+
 {{if (.ShowRole.HasRole "Owner")}}
-	<div class="ui basic label role-label">
+	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.owner_helper"}}">
 		{{ctx.Locale.Tr "repo.issues.owner"}}
 	</div>
+{{else if (.ShowRole.HasRole "Member")}}
+	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.member_helper"}}">
+		{{ctx.Locale.Tr "repo.issues.member"}}
+	</div>
+{{else if (.ShowRole.HasRole "Collaborator")}}
+	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.collaborator_helper"}}">
+		{{ctx.Locale.Tr "repo.issues.collaborator"}}
+	</div>
+{{else if and (.ShowRole.HasRole "First-time contributor") (.ShowRole.HasRole "Poster")}}
+	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.first_time_contributor_helper"}}">
+		{{ctx.Locale.Tr "repo.issues.first_time_contributor"}}
+	</div>
+{{else if (.ShowRole.HasRole "Contributor")}}
```

Comment:
I don't think it should be an `else if` but a normal `if` instead.
```suggestion
{{if (.ShowRole.HasRole "Member")}}
	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.member_helper"}}">
		{{ctx.Locale.Tr "repo.issues.member"}}
	</div>
{{end}}
{{if (.ShowRole.HasRole "Collaborator")}}
	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.collaborator_helper"}}">
		{{ctx.Locale.Tr "repo.issues.collaborator"}}
	</div>
{{end}}
{{if and (.ShowRole.HasRole "First-time contributor") (.ShowRole.HasRole "Poster")}}
	<div class="ui basic label role-label" data-tooltip-content="{{ctx.Locale.Tr "repo.issues.first_time_contributor_helper"}}">
		{{ctx.Locale.Tr "repo.issues.first_time_contributor"}}
	</div>
{{end}}
{{if (.ShowRole.HasRole "Contributor")}}
```

Reply:
Only show one role when the user have `Member`/`Owner`, `Collaborator` and `Contributor` in the same time.

Although this is done in the backend (for the performance), maybe it is better to also do this in the frontend?